lasagne/layers/input.py

```diff
@@ -67,6 +67,9 @@ def __init__(self, shape, input_var=None, name=None, **kwargs):
             if input_var.ndim != ndim:
                 raise ValueError("shape has %d dimensions, but variable has "
                                  "%d" % (ndim, input_var.ndim))
+            broad_pattern = [s == 1 for s in self.shape]
+            input_var = T.patternbroadcast(input_var, broad_pattern)
```

Comment:
I don't understand the problem here. You mean to say they both referred to the same object before(Prior to this PR) and now, they are different objects ? As in, if I understand the concern correctly, this does a deep copying whereas something like shallow copy is desired?